opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -2043,9 +2044,104 @@ namespace Opm
         }
     }
 
+    template <typename TypeTag>
+    void
+    StandardWell<TypeTag>::addWellPressureEquations(PressureMatrix& jacobian,
+                                                    const BVector& weights,
+                                                    const int pressureVarIndex,
+                                                    const bool use_well_weights,
+                                                    const WellState& well_state) const
+    {
+        // Add the well contributions in cpr
+        // use_well_weights is a quasiimpes formulation which is not implemented in multisegment
+        int bhp_var_index = Bhp;
+        int nperf = 0;
+        auto cell_weights = weights[0]*0.0;// not need for not(use_well_weights)
+        assert(this->duneC_.M() == weights.size());
+        const int welldof_ind = this->duneC_.M() + this->index_of_well_;
+        // do not assume anything about pressure controlled with use_well_weights (work fine with the assumtion also)
+        if(not(this->isPressureControlled(well_state)) || use_well_weights){
+            // make coupling for reservoir to well            
+            for (auto colC = this->duneC_[0].begin(), endC = this->duneC_[0].end(); colC != endC; ++colC) {
+                const auto row_ind = colC.index();
+                const auto& bw = weights[row_ind];
+                double matel = 0;        
+                assert((*colC).M() == bw.size());
+                for (size_t i = 0; i < bw.size(); ++i) {
+                    matel += (*colC)[bhp_var_index][i] * bw[i];
+                }
+                
+                jacobian[row_ind][welldof_ind] = matel;
+                cell_weights += bw;
+                nperf += 1;
+            }
+        }
+        cell_weights /= nperf;
+        
+        BVectorWell  bweights(1);
+        size_t blockSz = this->numWellEq_;
+        bweights[0].resize(blockSz);
+        bweights[0] = 0.0;
+        double diagElem = 0;
+        {            
```

Comment:
Nitpicks: extra block here seems unnecessary; missing space after if in lines 2087, 2111 and 2132; space around else in line 2109; extra spaces at end of many lines.